layer_metadata_restorer.go

```diff
@@ -13,29 +13,95 @@ import (
 
 //go:generate mockgen -package testmock -destination testmock/layer_metadata_restorer.go github.com/buildpacks/lifecycle LayerMetadataRestorer
 type LayerMetadataRestorer interface {
-	Restore(buildpacks []buildpack.GroupBuildpack, appMeta platform.LayersMetadata, cacheMeta platform.CacheMetadata) error
+	UseSHAFiles() bool
+	Restore(buildpacks []buildpack.GroupBuildpack, appMeta platform.LayersMetadata, cacheMeta platform.CacheMetadata, layerSHAStore LayerSHAStore) error
 }
 
 type DefaultLayerMetadataRestorer struct {
-	Logger     Logger
-	LayersDir  string
-	SkipLayers bool
+	logger      Logger
+	layersDir   string
+	skipLayers  bool
+	useShaFiles bool
 }
 
-func NewLayerMetadataRestorer(logger Logger, layersDir string, skipLayers bool) LayerMetadataRestorer {
+type LayerSHAStore interface {
```

Comment:
Could we add a docs comment here explaining how this "store" is using the filesystem OR memory?

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/624/commits/76b95b3821257dd6127e05958f303dcab437fab8. Personally I think that a comment is unnecessary now but please let me know if you think otherwise.